### DISCUSSION TITLE
About feature_importance

### DISCUSSION THREAD

**Discussion by mipypf** [pycaret/pycaret#1126] [Q&A] [answered]
Hi All,

Always thank you for your help. 
Excuse me for being a question of continuous throwing.

Can you help me understand the following questions?

[Question]
Is it correct to recognize that the calculation method of "Feature Importance" in "pycaret.regression.plot_model" is calculated from the "coef_" attribute or "feature_importance_" attribute of each estimator? Also, it would be helpful if you could tell me the code part to refer to.

Very much appreciate your support

**Comment by Yard1** [comment, marked as answer]
Those are indeed the attributes used. You can find the code here: https://github.com/pycaret/pycaret/blob/37f458b262f02f9abaa3d8f3c9dea25292c4fda1/pycaret/internal/tabular.py#L7153-L7194

**Reply by mipypf** [reply]
Thanks for the quick response.

For example, was it okay to understand how to calculate the "feature_importance_" attribute by each referenced machine learning library, such as Catboost, XGBoost, RandomForest (scikit-learn), etc.? I am studying, so I would appreciate it if you could teach me.

Very much appreciate your support

**Reply by Yard1** [reply]
I am not sure what you are asking for here. Could you clarify? Thanks.

**Reply by mipypf** [reply]
I'm sorry for the unclear question.

I think there are several ways to calculate the feature importance such as gain, division, weight, cover, prediction values ​​change, but how does pycaret calculate the feature importance of each estimator in "pycaret.regression.create_model"?

 I am studying, so I would appreciate it if you could teach me.
Very much appreciate your support

**Reply by Yard1** [reply]
The feature importances are obtained from `feature_importances_` and `coef_` attributes. How the values in those are calculated varies from model to model and is not determined by PyCaret.

**Reply by mipypf** [reply]
Thanks for the quick response.

I get the point.

Thank you for your continued support.
